### PR TITLE
feat: customize recent activity access

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
Adds a config flag `ENABLE_BROAD_ACTIVITY_ACCESS` that makes it possible to disable access to other users' recent activity data, both via the API and the profile page. In addition, a security manager method `raise_for_user_activity_access` is introduced to make it possible to fine tune who can and can't access recent activity data.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
When the flag is set to false, trying to access other users' activity data returns a 403 which is displayed on the profile page tables with the message "Access to user activity data is restricted" (in the screenshot the user is viewing another user's profile page):
image

With the default settings, accessing other users' activity data is allowed (current behavior unchanged):
image

### TESTING INSTRUCTIONS
1. set `ENABLE_BROAD_ACTIVITY_ACCESS = False` in `superset_config.py`
2. login as `admin`
3. create a user `test`
3. go to `/superset/profile/test`
4. notice the message "Access to user activity data is restricted" in all activity tables
5. comment out `ENABLE_BROAD_ACTIVITY_ACCESS` flag in `superset_config.py`
6. go to `/superset/profile/test`
7.  notice that access is unrestricted

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/core.py [hunk 1]
```diff
@@ -1526,7 +1538,7 @@ def available_domains(self) -> FlaskResponse:  # pylint: disable=no-self-use
     def fave_dashboards_by_username(self, username: str) -> FlaskResponse:
         """This lets us use a user's username to pull favourite dashboards"""
         user = security_manager.find_user(username=username)
-        return self.fave_dashboards(user.get_id())
+        return self.fave_dashboards(int(user.get_id()))
```
Comment (villebro) [inline superset/views/core.py:1541]: `user.get_id()` returns a string, so this should be cast to an int.

File: superset/security/manager.py [hunk 2]
```diff
@@ -1160,6 +1160,18 @@ def get_rls_ids(self, table: "BaseDatasource") -> List[int]:
         ids.sort()  # Combinations rather than permutations
         return ids
 
+    @staticmethod
+    def raise_for_user_activity_access(user_id: int) -> None:
+        id_ = int(g.user.get_id())
```
Comment (dpgaspar) [inline superset/security/manager.py:1165]: this may fail, `g.user` is not always safe

Comment (dpgaspar) [reply superset/security/manager.py:1165]: also, why not just `g.user.id` ?

File: superset/views/core.py (hunk 1 above)
Comment (dpgaspar) [inline superset/views/core.py:1541]: `user.id`?

File: superset/security/manager.py (hunk 2 above)
Comment (villebro) [reply superset/security/manager.py:1165]: Good point about `g.user`, will address. And good question about `g.user.get_id()` - I just noticed `g.user.get_id()` is used fairly prevalently, so I assumed it was the preferred way. `user.id` sounds better here IMO

File: UPDATING.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ assists people when migrating to a new version.
 
 ### Other
 
+- [17589](https://github.com/apache/incubator-superset/pull/17589): It is now possible to limit access to usersä recent activity data by setting the `ENABLE_PUBLIC_ACTIVITY_ACCESS` config flag to false, or customizing the `raise_for_user_activity_access` method in the security manager.
```
Comment (kgabryje) [inline UPDATING.md:45]: Typo - usersä 🙂

File: superset-frontend/src/components/TableLoader/TableLoader.test.tsx [hunk 4]
```diff
@@ -32,6 +32,7 @@ fetchMock.get('glob:*/api/v1/mock', [
 const defaultProps: TableLoaderProps = {
   dataEndpoint: '/api/v1/mock',
   addDangerToast: jest.fn(),
+  noDataText: 'No data available',
```
Comment (kgabryje) [inline superset-frontend/src/components/TableLoader/TableLoader.test.tsx:35]:
```suggestion
  noDataText: t('No data available'),
```

**Review by kgabryje — APPROVED** [review]
2 minor nits, otherwise looks and works great!

File: UPDATING.md (hunk 3 above)
Comment (villebro) [reply UPDATING.md:45]: Lol, that's what you get jumping between FIN and US keyboard layouts 1000x per day 😆

File: superset-frontend/src/components/TableLoader/TableLoader.test.tsx (hunk 4 above)
Comment (kgabryje) [reply superset-frontend/src/components/TableLoader/TableLoader.test.tsx:35]: Nevermind, I didn't notice it's a test 🤦

**Review by geido — APPROVED** [review]
LGTM!

**Pull request merged**